### DISCUSSION TITLE
About simplify

### DISCUSSION THREAD

**Discussion by gaoxipeng** [shapely/shapely#1911] [Q&A]
I need to simplify a MULTIPOLYGON,
use  simplify(geometry, tolerance, preserve_topology=True, **kwargs)
This uses the Douglas-Pucker algorithm. For each closed polygon, a starting and ending point needs to be selected. For the same shape, selecting different starting and ending points will result in different results. What are the principles for selecting these points? Can users be allowed to specify starting and ending points?
I tried to access the source code, but I just got    ： 
if preserve_topology:
        return lib.simplify_preserve_topology(geometry, tolerance, **kwargs)
    else:
        return lib.simplify(geometry, tolerance, **kwargs)

Where can I access the source code of this function? Are there any relevant papers available for reference?

**Comment by theroggy** [comment]
Shapely uses the [GEOS](https://libgeos.org) library under the hood, so if you want to know the algorythm you can check out the GEOS source code:

> Shapely is a Python package for set-theoretic analysis and
> manipulation of planar features using functions from the well known
> and widely deployed GEOS library. GEOS, a port of the Java Topology
> Suite (JTS), is the geometry engine of the PostGIS spatial extension
> for the PostgreSQL RDBMS. The designs of JTS and GEOS are largely
> guided by the Open Geospatial Consortium’s Simple Features Access
> Specification [1] and Shapely adheres mainly to the same set of
> standard classes and operations.

Source: [shapely user manual, introduction](https://shapely.readthedocs.io/en/stable/manual.html)
